allennlp/data/dataset_readers/wikitables/world.py

```diff
@@ -0,0 +1,111 @@
+from typing import List
+from pyparsing import OneOrMore, nestedExpr
+
+from nltk.sem.logic import LogicParser, Expression
+
+from allennlp.data.dataset_readers.wikitables import WTD, TableKnowledgeGraph
+
+
+class World:
+    """
+    We store all the information related to a world (i.e. the context in which logical forms will be
+    executed) here. For WikiTableQuestions, this includes a representation of a table, mapping from
+    Sempre variables in all logical forms to NLTK variables, and the types of all predicates and entities.
+    """
+    def __init__(self, table_filename: str) -> None:
+        self.table_graph = TableKnowledgeGraph.read_table_from_tsv(table_filename)
+        # NLTK has a naming convention for variable types (see ``_map_name`` for more details).
+        # We initialize this dict with common predicate names and update it as we process logical forms.
+        # TODO: Should we do updates while reading tables instead?
+        self.name_mapping = {"reverse": "R", "max": "M0", "min": "M1",
+                             "argmax": "A0", "argmin": "A1",
+                             "and": "A", "or": "O", "next": "N",
+                             "fb:cell.cell.date": "D", "fb:cell.cell.number": "B"}
+        # For every new Sempre column name seen, we update this counter to map it to a new NLTK name.
+        self._column_counter = 0
+        # Initial type signature. Will update when we see more predicates.
+        self.type_signature = {"R": WTD.REVERSE_TYPE, "M0": WTD.MAX_MIN_TYPE, "M1": WTD.MAX_MIN_TYPE,
+                               "A0": WTD.MAX_MIN_TYPE, "A1": WTD.MAX_MIN_TYPE}
+        self._logic_parser = LogicParser(type_check=True)
+
+    def process_sempre_forms(self, sempre_forms: List[str]) -> List[Expression]:
+        """
+        Processes logical forms in Sempre format (that are either gold annotations or output from DPD)
+        and converts them into ``Expression`` objects in NLTK. The conversion involves mapping names to
+        follow NLTK's variable naming conventions.
+        """
+        expressions = []
+        for sempre_form in sempre_forms:
+            parsed_lisp = OneOrMore(nestedExpr()).parseString(sempre_form).asList()
```

Comment:
I think bare functions are better done like `pyparsing.nestedExpr()` - it's more obvious where the function is coming from, instead of wondering where `nestedExpr()` is defined.

Reply:
Done.